src/main/java/se/sundsvall/myrepresentative/service/Mapper.java

```diff
@@ -0,0 +1,47 @@
+package se.sundsvall.myrepresentative.service;
+
+import org.springframework.stereotype.Component;
+import se.sundsvall.myrepresentative.api.model.CreateMandate;
+import se.sundsvall.myrepresentative.api.model.GranteeDetails;
+import se.sundsvall.myrepresentative.api.model.GrantorDetails;
+import se.sundsvall.myrepresentative.api.model.MandateDetails;
+import se.sundsvall.myrepresentative.api.model.MandateStatus;
+import se.sundsvall.myrepresentative.integration.db.entity.MandateEntity;
+
+@Component
+public class Mapper {
+
+	public MandateEntity toMandateEntity(final String municipalityId, final String namespace, final CreateMandate mandate) {
+		return new MandateEntity()
+			.withMunicipalityId(municipalityId)
+			.withGrantorPartyId(mandate.grantorDetails().grantorPartyId())
+			.withSignatoryPartyId(mandate.grantorDetails().signatoryPartyId())
+			.withGrantee(mandate.granteeDetails().partyId())
+			.withName(mandate.grantorDetails().name())
+			.withActiveFrom(mandate.activeFrom())
+			.withInactiveAfter(mandate.inactiveAfter())
+			.withNamespace(namespace);
+	}
+
+	public MandateDetails toMandateDetails(final MandateEntity mandateEntity) {
+		return new MandateDetails(
```

Comment:
Why not use the handy `@Builder`-annotation that you've added to the API record classes (`MandateDetailsBuilder.create()....build()`) 🙂?